source/jormungandr/jormungandr/scenarios/new_default.py

```diff
@@ -817,25 +818,21 @@ def fill_journeys(self, request_type, api_request, instance):
         journey_filter.final_filter_journeys(responses, instance, api_request)
         pb_resp = merge_responses(responses)
 
+        sort_journeys(pb_resp, instance.journey_order, api_request['clockwise'])
+        compute_car_co2_emission(pb_resp, api_request, instance)
+        tag_journeys(pb_resp)
+
         if 'ridesharing' in ridesharing_req['origin_mode'] and instance.ridesharing_services:
-            period_extremity = PeriodExtremity(ridesharing_req['datetime'], ridesharing_req['clockwise'])
+            logging.getLogger(__name__).debug('trying to add ridesharing journeys')
             try:
-                rs_journey, rs_tickets = build_ridesharing_crowfly_journey(instance,
-                                                                       ridesharing_req['origin'],
-                                                                       ridesharing_req['destination'],
-                                                                       period_extremity)
-                if rs_journey:
-                    pb_resp.journeys.extend([rs_journey])
-                    if rs_tickets:
-                        pb_resp.tickets.extend(rs_tickets)
-                    if pb_resp.HasField(str('error')) and not to_be_deleted(rs_journey):
-                        pb_resp.ClearField(str('error'))
+                decorate_journeys(pb_resp, instance, api_request)
             except:
                 logger.exception('Error while retrieving ridesharing ads')
+        else:
```

Comment:
(good point, I hesitated to point this out, i didn't, because this may complicate things...) I'm more interested in which cases this could happen?  The only case I can see is `ridesharing` is requested and no  ridesharing_services are available, right? If so, I'm wondering what the user will be waiting for when he put `ridesharing` in the requests intentionally? An explicit message like 'no matching ridesharing found' in the reponse? Or the user doesn't care?

Reply:
@xlqian yep, that's the case... short-circuit it would endup in a little bit of a 🍝 -code (`if`s dispatched for one functionnal purpose), but that's the only idea I got if we want to avoid that useless kraken call.